### PR TITLE
SSCSCI-482 - Remove email ID and telephone numbers when sending notification

### PR DESCRIPTION
**Pull request by gemmatalbot**
### Jira link (if applicable) ### 
https://tools.hmcts.net/jira/browse/SSCSCI-482

### Change description ### 
Remove email ID and telephone numbers when sending notification
]

### CODE REVIEW

**Review by NHAL22 — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsControllerTest.java [hunk 1]
```diff
@@ -66,13 +97,39 @@ void testPostRequestServiceHearingValues() throws Exception {
 
         given(serviceHearingsService.getServiceHearingValues(request))
             .willReturn(ServiceHearingValues.builder()
+                .caseDeepLink(null)
+                .caseManagementLocationCode("")
+                .caseRestrictedFlag(true)
+                .caseSlaStartDate("")
+                .externalCaseReference("")
+                .hearingChannels(null)
```
Comment (benouaer) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsControllerTest.java:105]: do you have to include this if you're just setting it to null? you can just omit it. probably same with empty string, no?

File: src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java [hunk 2]
```diff
@@ -55,7 +58,37 @@ public ResponseEntity<ServiceHearingValues> serviceHearingValues(
 
             ServiceHearingValues model = serviceHearingsService.getServiceHearingValues(request);
 
-            log.info("serviceHearingValues response {}", model);
+            var modelForLogging =  List.of(
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java:61]: do we need to log everything? 
and it seems like the only thing that needs masking is the party details, so why not get a copy of the model object, and just change the party details

Comment (gemmatalbot) [reply src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java:61]: spoke to nitin who asked to keep everything excpet PII, we removed names from serviceHearingValues, and then names, email address and phone from partyDetails

File: src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java [hunk 3]
```diff
@@ -102,4 +134,31 @@ private void logException(Exception exc, String caseId) {
             log.error("Error updating case id {}, {}", caseId, exc);
         }
     }
+
+    private List getPartyDetailsForLog(List<PartyDetails> partyDetails) {
```
[the remark this answers is not in the document]
Comment (gemmatalbot) [reply src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java:138]: As above, spoke to Nitin and kept everything except removed the values containing PII

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsControllerTest.java (hunk 1 above)
Comment (gemmatalbot) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsControllerTest.java]: true, I only need to keep the partyDetails part of it for code test coverage - have removed all the null/empty values :)

File: src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java [hunk 4]
```diff
@@ -102,4 +134,31 @@ private void logException(Exception exc, String caseId) {
...
+                    List.of(
+                        Optional.ofNullable(party.getPartyID()),
+                        Optional.ofNullable(party.getPartyType()),
+                        Optional.ofNullable(party.getPartyChannel()),
+                        Optional.ofNullable(party.getPartyRole()),
+                        Optional.ofNullable(party.getIndividualDetails().getPreferredHearingChannel()),
+                        Optional.ofNullable(party.getIndividualDetails().getInterpreterLanguage()),
+                        Optional.ofNullable(party.getIndividualDetails().getReasonableAdjustments()),
+                        Optional.ofNullable(party.getIndividualDetails().isVulnerableFlag()),
+                        Optional.ofNullable(party.getIndividualDetails().getVulnerabilityDetails()),
```
Comment (nitinprabhuhmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java:152]: this can contain sensitive Info ?

Comment (gemmatalbot) [reply src/main/java/uk/gov/hmcts/reform/sscs/controllers/ServiceHearingsController.java:152]: It didn't on the logs we observed, but that doesn't preclude it from happening so will remove to be safe :)

**Review by nitinprabhuhmcts — DISMISSED** [review]

**Review by NHAL22 — APPROVED** [review]

**Review by hashimalisolirius — APPROVED** [review]

**Review by benouaer — APPROVED** [review]
